contents/newsletter/choosing-technologies.md

```diff
@@ -0,0 +1,158 @@
+
+- Migrated more of our highest volume services from Python to Rust.
+
+- Evaluated Warpstream and Pulsar as alternatives to Kafka.
+
+- Adopted C++ over Python to speed up SQL parsing. 
+
+We've done all this while shipping multiple new products like web analytics, mobile session replay, and surveys as well as adding features to existing ones. Each of these comes with its own set of technologies to maintain and improve.
+
+Technology is a tool to help us build something people want. This relies on balancing choosing the **right** technology with shipping fast. Many companies focus too much on the first part and become bogged down in process.
+
+A core value of PostHog is **trust and feedback over process**. This means the  path we take to adopt a technology looks different each time. What stays the same is our reliance on our small teams and their expertise. 
+
+Although we can't tell you what the right technologies are or the perfect way to pick them, we **can** share how and why we rely on autonomy to choose them.
+
+## Start with a hair-on-fire problem then find an extinguisher
+
+Adopting new technologies is hard and expensive, so "nice to have" is not enough. We look for hair-on-fire problems, which typically means one of three things:
+
+1. **Costs.** Our infrastructure team does a quarterly review of AWS costs, evaluates where spend is going, and identifies areas to improve it. Costs rising above standard benchmarks is a trigger to look for solutions. 
+
+2. **Infrastructure.** When we reach scaling limits for our current infrastructure, and adding "more boxes" won't fix the issue, it's time to look for another solution. 
+
+3. **Customers.** When customers demand a new feature, like reliable and resumable data exports, we need to consider whether we build or buy.
+
+The goal of technology is to solve this specific hair-on-fire problem. We find potential technologies that do this in one of two ways: 
+
+1. **What are other people using to solve the problem?** Technologies are by definition tools or services other people have used to solve problems. One reason [we chose ClickHouse](/blog/how-we-turned-clickhouse-into-our-eventmansion), for example, was that Cloudflare used it.
+
+2. **How have YOU solved this problem in the past?** Pitfalls are often avoidable if you already fell in them once. This means we often rely on our teams' experience with a technology to guide our evaluation.
+
+Realistically, the evaluation process happens at the same time as figuring out the options. This is because our team knows some options can be written off immediately as they don't fit our requirements. When we were looking for [scalable replacements for Postgres](/blog/how-we-turned-clickhouse-into-our-eventmansion) in our early days, for example, we knew we needed it to be open source and self-hostable. This immediately cut down our list. 
+
+## Evaluate solutions autonomously
```

Comment:
```suggestion
## We evaluate solutions autonomously
```

Reply:
Alternatively, I wonder if we should focus on evaluating closer to the real world here?